lib/loc.mli

```diff
@@ -35,13 +32,11 @@ val unloc : t -> int * int
 val make_loc : int * int -> t
 (** Make a location out of its start and end position *)
 
-val ghost : t
-(** Dummy location *)
-
-val is_ghost : t -> bool
-(** Test whether the location is meaningful *)
-
 val merge : t -> t -> t
+val mergeo : t option -> t option -> t option
+
+val merge_opt : t option -> t -> t
+val opt_merge : t -> t option -> t
```

Comment:
I find these two names a bit too clever and hard to remember. Are those functions actually needed? There is no use of them in the patch, and it looks like `mergeo` (which could be named `merge_opt` in my book) may be enough.